packages/quantic/force-app/main/default/lwc/quanticResultAction/quanticResultAction.css

```diff
@@ -0,0 +1,40 @@
+.slds-nubbin_bottom-left:before,
+.slds-nubbin_bottom-left:after {
+  left: 50%;
+}
+
+.slds-popover_tooltip {
+  width: max-content;
+}
+
+.result-action_tooltip {
+  position: absolute;
+  top: -3rem;
+  left: 50%;
+  transform: translate(-50%, 0);
+}
+
+.result-action_container {
+  display: inline;
+}
+
+.result-action_container:hover > .result-action_tooltip {
+  visibility: visible;
+  opacity: 1;
+}
+
+.result-action_button {
+  --lwc-borderWidthThin: 0.5px;
+}
+
+.result-action_first {
+  --sds-c-button-radius-border: 4px 0px 0px 4px;
+}
+
+.result-action_middle {
+  --sds-c-button-radius-border: 0px;
+}
+
+.result-action_last {
+  --sds-c-button-radius-border: 0px 4px 4px 0px;
+}
```

Comment:
Is there a way to work with `:first-child` and `:last-child` instead here?

Something like this:

```
.result-actionList:first-child .result-action_button {
  --sds-c-button-radius-border: 4px 0px 0px 4px;
}

.result-actionList:last-child .result-action_button {
  --sds-c-button-radius-border: 0px 4px 4px 0px;
}
```

You'd have to be careful if there's only 1 button for example by instead of setting the class `result-action_button` you could give it the class `result-action_only-button` so you can give it both rounded corners (I don't know how first-child and last-child will behave if there's only 1 child element)

Reply:
The special thing about the `result-action_first` `result-action_middle` and `result-action_last` classes is that these classes need to be set directly on the `lightning-button-icon` and the `lightning-button-icon-stateful` components. these components are custom components from the Salesforce library and that's the only way to customize them following the documentation., that's why the css classes are created and used at the QuanticResultAction level.
 
Your suggestion means that these CSS rules and the `result-action_button` css class will be defined at the QuanticResultActionBar level, which is kinda problematic because the styles  can't cross the shadow DOM boundaries  and they will not be visible inside the `QuanticResultAction` component